tests/test_runner/test_discover_runner.py

```diff
@@ -58,6 +68,48 @@ def test_add_arguments_debug_mode(self):
         ns = parser.parse_args(["--debug-mode"])
         self.assertTrue(ns.debug_mode)
 
+    def test_setup_shuffler_no_shuffle_argument(self):
+        runner = DiscoverRunner()
+        self.assertIs(runner.shuffle, False)
+        shuffler = runner.setup_shuffler()
+        self.assertIsNone(runner.shuffle_seed)
+        self.assertIsNone(shuffler)
+
+    def test_setup_shuffler_shuffle_none(self):
+        # Prevent writing the seed to stdout.
+        runner = DiscoverRunner(shuffle=None)
+        self.assertIsNone(runner.shuffle)
+        with mock.patch('random.randint', return_value=1):
+            with captured_stdout() as stdout:
+                shuffler = runner.setup_shuffler()
+        self.assertEqual(stdout.getvalue(), 'Using shuffle seed: 1 (generated)\n')
+        self.assertEqual(runner.shuffle_seed, 1)
+        self.assertIsInstance(shuffler, Shuffler)
+        nums = list(range(10))
+        actual = shuffler.shuffle(nums, lambda x: x)
+        self.assertEqual(actual, [9, 1, 8, 7, 5, 4, 2, 3, 6, 0])
+
+    def test_setup_shuffler_shuffle_int(self):
+        cases = [
+            (0, [9, 5, 4, 1, 2, 0, 7, 3, 8, 6]),
+            (1, [9, 1, 8, 7, 5, 4, 2, 3, 6, 0]),
+            (2, [7, 4, 8, 3, 1, 6, 9, 5, 0, 2]),
+        ]
+        nums = list(range(10))
+        for shuffle, expected in cases:
+            with self.subTest(shuffle=shuffle):
+                # Prevent writing the seed to stdout.
```

Comment:
This comment seems incorrect. We don't set `verbosity` here and `stdout` is captured with `captured_stdout()`. I would remove it.